chia/wallet/puzzles/dao_spend_p2_singleton_v2.clsp

```diff
@@ -186,21 +192,31 @@
     (list ASSERT_MY_PUZZLEHASH (calculate_singleton_puzzle_hash TREASURY_SINGLETON_STRUCT treasury_inner_puzhash))
     (c
       (list CREATE_COIN treasury_inner_puzhash ONE (list (f (r TREASURY_SINGLETON_STRUCT))))
-      (loop_through_list
-        TREASURY_SINGLETON_STRUCT
-        (sum_create_coins CONDITIONS)
-        P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH
-        (calculate_coin_id (f (f p2_singleton_parent_amount_list)) P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH (f (r (f p2_singleton_parent_amount_list))))
-        (r p2_singleton_parent_amount_list)
-        (f (r (f p2_singleton_parent_amount_list)))
-        (for_each_asset_type
-          TREASURY_SINGLETON_STRUCT
-          CAT_MOD_HASH
-          P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH
-          LIST_OF_TAILHASH_CONDITIONS
-          p2_singleton_tailhash_parent_amount_list  ; ((tailhash ((parent amount) (parent_amount)... ) (tailhash (parent amount))..)
-          CONDITIONS
-        )
+      (if CONDITIONS
+          (loop_through_list
+            TREASURY_SINGLETON_STRUCT
+            (sum_create_coins CONDITIONS)
+            P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH
+            (if p2_singleton_parent_amount_list (calculate_coin_id (f (f p2_singleton_parent_amount_list)) P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH (f (r (f p2_singleton_parent_amount_list)))) ())
+            (if p2_singleton_parent_amount_list (r p2_singleton_parent_amount_list) ())
+            (if p2_singleton_parent_amount_list (f (r (f p2_singleton_parent_amount_list))) ())
+            (for_each_asset_type
+              TREASURY_SINGLETON_STRUCT
+              CAT_MOD_HASH
+              P2_SINGLETON_VIA_DELEGATED_PUZZLE_PUZHASH
+              LIST_OF_TAILHASH_CONDITIONS
+              p2_singleton_tailhash_parent_amount_list  ; ((tailhash ((parent amount) (parent_amount)... ) (tailhash (parent amount))..)
+              CONDITIONS
+            )
+          )
+          (for_each_asset_type
```

Comment:
should we disallow spending coins if there are no `CONDITIONS`?

Reply:
We need to capture the case where there are TAILHASH_CONDITIONS but not [xch] CONDITIONS. If neither are present then it will just return nil